src/conn/conn_oligarch.c

```diff
@@ -93,6 +88,63 @@ __disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t checkpoint_id)
     /* Put our new config in */
     WT_ERR(__wt_metadata_insert(internal_session, metadata_key, cfg_ret));
 
+    /*
+     * Part 2: Get the metadata for other tables from the shared metadata table.
+     */
+
+    /* We need a separate internal session to pick up the new checkpoint. */
+    WT_ERR(__wt_open_internal_session(
+      conn, "checkpoint-pick-up-shared", false, 0, 0, &shared_metadata_session));
+
+    /* Scan the metadata table. */
+    cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_open_cursor);
+    cfg[1] = "checkpoint=" WT_CHECKPOINT ",checkpoint_use_history=false";
+    cfg[2] = NULL;
+    WT_ERR(__wt_open_cursor(shared_metadata_session, WT_DISAGG_METADATA_URI, NULL, cfg, &cursor));
+
+    while ((ret = cursor->next(cursor)) == 0) {
+        WT_ERR(cursor->get_key(cursor, &metadata_key));
+        WT_ERR(cursor->get_value(cursor, &metadata_value));
+
+        md_cursor->set_key(md_cursor, metadata_key);
+        WT_ERR_NOTFOUND_OK(md_cursor->search(md_cursor), true);
+
+        if (ret == 0) {
+            /* Existing table: Just apply the new metadata. */
+            WT_ERR(__wt_config_getones(session, metadata_value, "checkpoint", &cval));
+            len = strlen("checkpoint=") + strlen(metadata_value) + 1 /* for NUL */;
+            if (len > metadata_value_cfg_len) {
+                metadata_value_cfg_len = len;
+                WT_ERR(
+                  __wt_realloc_noclear(session, NULL, metadata_value_cfg_len, &metadata_value_cfg));
+            }
+            WT_ERR(
+              __wt_snprintf(metadata_value_cfg, len, "checkpoint=%.*s", (int)cval.len, cval.str));
+
+            /* Merge the new checkpoint metadata into the current table metadata. */
+            WT_ERR(md_cursor->get_value(md_cursor, &current_value));
+            cfg[0] = current_value;
+            cfg[1] = metadata_value_cfg;
+            cfg[2] = NULL;
+            WT_ERR(__wt_config_collapse(session, cfg, &cfg_ret));
```

Comment:
Will we keep all the checkpoint in the metadata now?

Reply:
Yes, the shared metadata table keeps checkpoints for all shared tables, except for itself.

Reply:
Sorry, my question is do we keep all the checkpoints we do for a table in the metadata? The config collapse will overwrite the checkpoint config with the latest checkpoint or it will combine the latest checkpoint and the previous checkpoint and store them together?

Reply:
It will overwrite it. Here is an output from the table metadata for a table that was checkpointed several time—note that there is only one checkpoint info available:

```
6: desc: 0x7d short key 31 bytes:
  "file:test_oligarch06.wt_stable"
7: desc: 0x80 val 1472 bytes:
"access_pattern_hint=none,allocation_size=4KB,app_metadata=,assert=(commit_timestamp=none,durable_timestamp=none,read_timestamp=none,write_timestamp=off),block_allocation=best,block_compressor=,block_manager=default,cache_resident=false,checkpoint=(WiredTigerCheckpoint.160=(addr="ce4d8380c077e41ec6ee21",order=160,time=1731888191,size=3725,newest_start_durable_ts=0,oldest_start_ts=0,newest_txn=1200316,newest_stop_durable_ts=0,newest_stop_ts=-1,newest_stop_txn=-11,prepare=0,write_gen=3662,run_write_gen=1,next_page_id=3726)),checkpoint_backup_info=,checkpoint_lsn=,checksum=on,collator=,columns=,dictionary=0,disaggregated=(page_log=palm,stable_prefix=.,storage_source=),encryption=(keyid=,name=),format=btree,huffman_key=,huffman_value=,id=4,ignore_in_memory_cache_size=false,in_memory=false,internal_item_max=0,internal_key_max=0,internal_key_truncate=true,internal_page_max=4KB,key_format=S,key_gap=10,leaf_item_max=0,leaf_key_max=0,leaf_page_max=32KB,leaf_value_max=0,log=(enabled=true),memory_page_image_max=0,memory_page_max=5MB,oligarch_log=(enabled=false,oligarch_constituent=false),os_cache_dirty_max=0,os_cache_max=0,prefix_compression=false,prefix_compression_min=4,readonly=false,split_deepen_min_child=0,split_deepen_per_child=0,split_pct=90,tiered_object=false,tiered_storage=(auth_token=,bucket=,bucket_prefix=,cache_directory=,local_retention=300,name=,object_target_size=0),value_format=S,verbose=[],version=(major=2,minor=1),write_timestamp_usage=none"
```

This is the same mechanism that we were using so far, except for just a single table, so there is no change in that sense.